src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/ReviewUnusedParameters.Fixer.cs

```diff
@@ -22,9 +30,98 @@ public sealed override FixAllProvider GetFixAllProvider()
+                }
+
+                solution = solution.WithDocumentSyntaxRoot(group.Key, editor.GetChangedRoot());
+            }
+
+            return solution;
+        }
+
+        private async Task<ImmutableArray<KeyValuePair<DocumentId, SyntaxNode>>> GetNodesToRemoveAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            SyntaxNode root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            SyntaxNode node = root.FindNode(diagnostic.Location.SourceSpan);
+            node = GetParameterNode(node);
+            var nodesToRemove = ImmutableArray.CreateBuilder<KeyValuePair<DocumentId, SyntaxNode>>();
+            nodesToRemove.Add(new KeyValuePair<DocumentId, SyntaxNode>(document.Id, node));
+
+            DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            ISymbol parameterSymbol = editor.SemanticModel.GetDeclaredSymbol(node);
+            ISymbol methodDeclarationSymbol = editor.SemanticModel.GetDeclaredSymbol(node.Parent.Parent);
+            var referencedSymbols = await SymbolFinder.FindReferencesAsync(methodDeclarationSymbol, document.Project.Solution, cancellationToken).ConfigureAwait(false);
+
+            foreach (var referencedSymbol in referencedSymbols)
+            {
+                if (referencedSymbol.Locations != null)
+                {
+                    foreach (var referenceLocation in referencedSymbol.Locations)
+                    {
+                        Location location = referenceLocation.Location;
+                        var referencedSymbolNode = location.SourceTree.GetRoot().FindNode(location.SourceSpan).Parent;
+                        referencedSymbolNode = GetOperationNode(referencedSymbolNode);
+                        DocumentEditor localEditor = await DocumentEditor.CreateAsync(referenceLocation.Document, cancellationToken).ConfigureAwait(false);
+                        var operation = localEditor.SemanticModel.GetOperation(referencedSymbolNode, cancellationToken);
 
+                        var arguments = (operation as IObjectCreationOperation)?.Arguments;
+                        if (arguments == null)
+                        {
+                            arguments = (operation as IInvocationOperation)?.Arguments;
+                        }
+
+                        if (arguments != null)
+                        {
+                            foreach (IArgumentOperation argument in arguments)
+                            {
+                                if (argument.Parameter.Equals(parameterSymbol) && (argument.ArgumentKind == ArgumentKind.Explicit))
+                                {
+                                    nodesToRemove.Add(new KeyValuePair<DocumentId, SyntaxNode>(referenceLocation.Document.Id, referencedSymbolNode.FindNode(argument.Syntax.GetLocation().SourceSpan)));
```

Comment:
I think we need to handle a bunch of cases here where removing the argument node can lead to side effects and bail out.